docs/batch.md

```diff
@@ -0,0 +1,108 @@
+# Batched Requests in Apollo Fetch
+
+`ApolloFetch` supports batched GraphQL requests with middleware and afterware without any additional work.
+
+## Usage
+
+`createApolloFetch` returns a function that implements the `ApolloFetch` interface, which overloads the function to accept an array of `GraphQL` requests in addition to the standard single request.
+This example sends `request` twice with a single network request.
+
+```js
+const apolloFetch = createApolloFetch({ uri });
+const request = { query, variables, operationName };
+
+apolloFetch([request, request])
+  .then(results => {
+    results.map(({ data, error, extensions }) => {
+      //access each GraphQL result independently
+    });
+  })
+  .catch(error => {
+    //respond to a network error
+  });
+```
+
+Batched middleware and afterware are added to an `ApolloFetch` with `batchUse` and `batchUseAfter`.
+Middleware has access to the array of requests in `request`.
+Afterware has access to the single network response, which contains the array of GraphQL results in `parsed`.
+
+```js
+const apolloFetch = createApolloFetch();
+
+const batchMiddleware = ({ requests, options }, next) => { ... next(); };
+
+const batchAfterware = ({ response, options }, next) => { ... next(); };
+
+apolloFetch.batchUse(batchMiddleware);
+apolloFetch.batchUseAfter(batchAfterware);
+```
+
+Batched middleware and afterware exhibit the same chaining semantics as the single request variant.
+
+```js
+const apolloFetch = createApolloFetch();
+apolloFetch
+  .batchUse(middleware1)
+  .batchUse(middleware2)
+  .batchUseAfter(afterware1)
+  .batchUseAfter(afterware2)
+  .batchUse(middleware3);
+```
+
+### Batch Formatting
+
+Since GraphQL does not include batched requests in the formal spec, `apollo-fetch` creates a request that is compatible with [`apollo-server`](https://github.com/apollographql/apollo-server).
```

Comment:
> GraphQL does not include batched requests in the formal spec

No kind of request is mentioned in the spec. So no need to make this exception here.

Maybe say "not all GraphQL servers support batching out of the box" or something.